tools/job_metadata.py

```diff
@@ -21,13 +21,53 @@
 # (none yet)
 
 
+# the job's working directory (JWD) and subdirectories may contain various
+# files storing metadata for a job
+# below, we define constants for sections and 'settings' in these files
+#
+# job config directory name and filename
+JOB_CFG_DIRECTORY_NAME = "cfg"
+JOB_CFG_FILENAME = "job.cfg"
+
+# JWD/cfg/$JOB_CFG_FILENAME
+JOB_CFG_ARCHITECTURE_SECTION = "architecture"
+JOB_CFG_ARCHITECTURE_OS_TYPE = "os_type"
+JOB_CFG_ARCHITECTURE_SOFTWARE_SUBDIR = "software_subdir"
+
+JOB_CFG_REPOSITORY_SECTION = "repository"
+JOB_CFG_REPOSITORY_CONTAINER = "container"
+JOB_CFG_REPOSITORY_REPOS_CFG_DIR = "repos_cfg_dir"
+JOB_CFG_REPOSITORY_REPO_ID = "repo_id"
+JOB_CFG_REPOSITORY_REPO_NAME = "repo_name"
+JOB_CFG_REPOSITORY_REPO_VERSION = "repo_version"
+
+JOB_CFG_SITE_CONFIG_SECTION = "site_config"
+JOB_CFG_SITE_CONFIG_BUILD_LOGS_DIR = "build_logs_dir"
+JOB_CFG_SITE_CONFIG_CONTAINER_CACHEDIR = "container_cachedir"
+JOB_CFG_SITE_CONFIG_HTTP_PROXY = "http_proxy"
+JOB_CFG_SITE_CONFIG_HTTPS_PROXY = "https_proxy"
+JOB_CFG_SITE_CONFIG_LOAD_MODULES = "load_modules"
+JOB_CFG_SITE_CONFIG_LOCAL_TMP = "local_tmp"
+JOB_CFG_SITE_CONFIG_SHARED_FS_PATH = "shared_fs_path"
+
+# JWD/_bot_jobJOBID.metadata
 JOB_PR_SECTION = "PR"
+JOB_PR_REPO = "repo"
+JOB_PR_PR_NUMBER = "pr_number"
+JOB_PR_PR_COMMENT_ID = "pr_comment_id"
+
+# JWD/_bot_jobJOBID.result
+JOB_RESULT_SECTION = "RESULT"
 JOB_RESULT_ARTEFACTS = "artefacts"
+JOB_RESULT_COMMENT_DESCRIPTION = "comment_description"
 JOB_RESULT_FAILURE = "FAILURE"
-JOB_RESULT_SECTION = "RESULT"
 JOB_RESULT_STATUS = "status"
 JOB_RESULT_SUCCESS = "SUCCESS"
```

Comment:
Should this not be lower case as it is part of the RESULT section?
```suggestion
JOB_RESULT_SUCCESS = "success"
```

Reply:
This is maybe a bit confusing, because some constants represent the names of settings (in a `_bot_jobJOBID.result` file): `JOB_RESULT_ARTEFACTS`, `JOB_RESULT_COMMENT_DESCRIPTION` and `JOB_RESULT_STATUS`.

Other constants (`JOB_RESULT_FAILURE` and `JOB_RESULT_SUCCESS`) represent values of the setting `JOB_RESULT_STATUS`. These values are determined by the `bot/check-build.sh` script and then written to a `_bot_jobJOBID.result` file.

I restructured the definition of the constants and added some comments to explain what they are used for.

See commit https://github.com/EESSI/eessi-bot-software-layer/commit/d3b211c405e1b4be64fc467457fb5425773393a4